multimap/multimap.go

```diff
@@ -0,0 +1,48 @@
+// Package multimap provides an associative container that permits multiple entries with the same key.
+//
+// There are four implementations of the MultiMap data structure, identified by separate New* functions.
+// They differ in the following ways:
+//  - whether key type and value type must be comparable.
+//  - whether duplicate entries (same key and same value) are permitted.
+//  - whether keys and values are sorted or unsorted in Get, Each, and EachAssociation methods.
+package multimap
+
+// MultiMap is an associative container that contains a list of key-value pairs, while permitting multiple entries with the same key.
+type MultiMap[K, V any] interface {
+	// Dimension returns number of distinct keys.
+	Dimension() int
+	// Size returns total number of entries.
+	Size() int
+
+	// Count returns number of entries with a given key.
+	Count(key K) int
+	// Has determines whether at least one entry exists with a given key.
+	Has(key K) bool
+	// Get returns a list of values with a given key.
+	Get(key K) []V
+
+	// Set adds an entry.
+	// Whether duplicate entries are allowed depends on the chosen implementation.
+	Set(key K, value V)
```

Comment:
Perhaps to be consistent with the other data structures this function should be called `Put`.